tools/codegen/config.yml

```
@@ -146,23 +141,197 @@ resources:
       aliases:
         tenant_name: name # path param name does not match the API request property
 
-  # No fully functional yet so it doesn't have acceptance tests yet.
   cluster_api:
     read:
       path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}
       method: GET
     create:
       path: /api/atlas/v2/groups/{groupId}/clusters
       method: POST
+      wait: &cluster-create-wait
+        state_property: "stateName"
+        pending_states:
+          [
+            "CREATING",
+            "UPDATING",
+            "REPAIRING",
+            "REPEATING",
+            "PENDING",
+            "DELETING",
+          ]
+        target_states: ["IDLE"]
+        timeout_seconds: 10800 # 3 hours
+        min_timeout_seconds: 60
+        delay_seconds: 30
     update:
       path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}
       method: PATCH
+      wait: *cluster-create-wait
     delete:
       path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}
       method: DELETE
+      wait:
+        <<: *cluster-create-wait
+        pending_states:
+          [
+            "IDLE",
+            "CREATING",
+            "UPDATING",
+            "REPAIRING",
+            "REPEATING",
+            "PENDING",
+            "DELETING",
+          ]
+        target_states: ["DELETED"] # DELETED is a special state value when API returns 404 or empty object
     schema:
       aliases:
         cluster_name: name # path param name does not match the API request property
+      # these attributes are changed by Atlas so need to be optional
```

Just making it super clear for future reference
```suggestion
      # These attributes are optional but when not configured Atlas returns a value
      #  If a nest attribute is O/C, all its optional child attributes must also be defined as O/C to avoid non empty plans
```